dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/Category.java

```diff
@@ -64,6 +65,13 @@
 
     private List<CategoryCombo> categoryCombos = new ArrayList<>();
 
+    /**
+     * This list holds CategoryOption's which the current user is not allowed to
+     * read. It's, currently, populated and used during Events queries. See
+     * {@link org.hisp.dhis.analytics.security.DefaultAnalyticsSecurityManager#extractNonAuthorizedCategoriesOptionsFrom(List)}
+     */
+    private transient List<CategoryOption> nonAuthorizedCategoryOptions = new ArrayList<>();
```

Comment:
I would prefer not to "pollute" the data model with this transient List. It is specific to analytics and does not belong naturally at the category domain entity.

Reply:
I agree, I just couldn't find a better place while avoiding passing in another object until the JdbcEventAnalyticsManager. My comment above, to Luciano, has some approaches, but they don't look good either. Do you like some of the approaches suggested in my previous comments or has another suggestion? Remember that we cannot manipulate any object in Category, otherwise, it will be persisted.